src/roadmap/v1/lifecycle/rhel.py

```diff
@@ -1,47 +1,185 @@
-async def get_systems_major(major: int = Path(..., description="Major version number")):
-    systems = get_systems_data(major)
+class LifecycleResponse(BaseModel):
+    data: list[RHELLifecycle]
 
-    return sorted(systems, key=itemgetter("major", "minor"), reverse=True)
+
+@router.get("/", summary="Return lifecycle data for all RHEL versions", response_model=LifecycleResponse)
+async def get_systems():
+    return {"data": get_lifecycle_data()}
 
 
-@router.get("/{major}/{minor}")
+@router.get("/{major}", response_model=LifecycleResponse)
+async def get_systems_major(
+    major: MajorVersion,
+):
+    return {"data": get_lifecycle_data(major)}
+
+
+@router.get("/{major}/{minor}", response_model=LifecycleResponse)
 async def get_systems_major_minor(
-    major: int = Path(..., description="Major version number"),
-    minor: int = Path(..., description="Minor version number"),
+    major: MajorVersion,
+    minor: MinorVersion,
 ):
-    systems = get_systems_data(major, minor)
+    return {"data": get_lifecycle_data(major, minor)}
+
 
-    return sorted(systems, key=itemgetter("major", "minor"), reverse=True)
+def get_lifecycle_data(major: int | None = None, minor: int | None = None, reverse: bool = True):
+    lifecycles = (item for item in OS_LIFECYCLE_DATES.values() if item.minor is not None)
 
+    if major and minor is not None:
+        lifecycles = (item for item in lifecycles if (item.major, item.minor) == (major, minor))
+    elif major:
+        lifecycles = (item for item in lifecycles if item.major == major)
 
-def get_systems_data(major=None, minor=None):
-    data = OS_DATA_MOCKED
+    return sorted(lifecycles, key=attrgetter("major", "minor"), reverse=reverse)
+
+
+## Relevant ##
+relevant = APIRouter(
+    prefix="/relevant/lifecycle/rhel",
+    tags=["Relevant", "RHEL"],
+)
 
-    if major is not None:
-        data = [d for d in data if d["major"] == major]
-    if minor is not None:
-        data = [d for d in data if d["minor"] == minor]
 
-    return data
+@relevant.get("/{major}/{minor}")
+@relevant.get("/{major}")
+@relevant.get("/")
+async def get_relevant_systems(
+    authorization: t.Annotated[str | None, Header(include_in_schema=False)] = None,
+    user_agent: t.Annotated[str | None, Header(include_in_schema=False)] = None,
+    x_rh_identity: t.Annotated[str | None, Header(include_in_schema=False)] = None,
+    major: int | None = None,
+    minor: int | None = None,
+) -> RelevantSystemsResponse:
+    headers = {
+        "Authorization": authorization,
+        "User-Agent": user_agent,
+        "X-RH-Identity": x_rh_identity,
+    }
+    systems_response = await get_system_count_from_inventory(headers=headers, major=major, minor=minor)
+
+    system_counts = defaultdict(int)
+    for result in systems_response.get("results", []):
+        system_profile = result["system_profile"]
+        if not system_profile:
+            logger.info("Unable to get relevant systems due to missing system profile")
```

Comment:
if you have the system's ID here, print it so we are able to find the offending record

Reply:
I put this in the log message for now. Ideally ID would be a separate field, but we'll have to do that when logging gets overhauled.